Code/Framework/AzToolsFramework/AzToolsFramework/ComponentModes/ShapeComponentModeBus.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Component/ComponentBus.h>
+
+namespace AzToolsFramework
+{
+    //! Bus used to communicate with shape component mode.
+    class ShapeComponentModeRequests
+        : public AZ::EntityComponentBus
+    {
+    public:
+        enum class SubMode : AZ::u32
+        {
+            Dimensions,
+            TranslationOffset,
+            NumModes
+        };
+
+        //! Gets the current shape component mode sub mode.
+        virtual SubMode GetCurrentMode() = 0;
+
+        //! Sets the current shape component mode sub mode.
+        //! @param mode The new sub mode to set.
+        virtual void SetCurrentMode(SubMode mode) = 0;
```

Comment:
suggestion: `GetCurrentMode` and `SetCurrentMode` are quite generic and almost certainly already exist in the codebase in another context. For a bus/interface it might be worth being slightly more specific (e.g. `GetShapeSubMode` or even `GetShapeComponentModeSubMode` (though maybe that's too verbose looking at it now 🙈). It's just a shame 'Mode' is used all over the place and can make finding references/navigating in future more difficult.